pexpect/spawnbase.py

```diff
@@ -88,7 +92,11 @@ def __init__(self, timeout=30, maxread=2000, searchwindowsize=None,
             self.string_type = bytes
             self.buffer_type = BytesIO
             self.crlf = b'\r\n'
-            if PY3:
+            try:               # Python 2
+                self.allowed_string_types = (basestring, )
+                self.linesep = os.linesep
+                self.write_to_stdout = sys.stdout.write
+            except NameError:  # Python 3
```

Comment:
I strongly prefer the `if PY3` check here. Catching NameError is a bug waiting to happen. If we can't find a way to use flake8 with the explicit `if PY3`, I'd rather not use it at all.

Reply:
To clarify: I'd rather not use it automated in CI. It's still fine to run it manually and fix problems.